lib/Controller/ApiController.php

```diff
@@ -93,10 +94,18 @@ public function setDownloadLimit(string $token, int $limit): Response {
 
 		// Update DB
 		$shareLimit->setLimit($limit);
-		if ($insert) {
-			$this->mapper->insert($shareLimit);
+		if($limit == 0 || $limit == ''){
```

Comment:
```suggestion
		if (($limit ?? 0) ===  0) {
```

Reply:
Updated the changes in new commit. Please take a look.

Thanks :)